test/util/tile_cover.test.cpp

```diff
@@ -85,11 +85,167 @@ TEST(TileCover, SanFranciscoZ0Wrapped) {
+        { 15,9373,12536 },
+        { 15,9374,12536 },
+        { 15,9373,12537 },
+        { 15,9374,12537 },
+        }));
+}
+
+TEST(TileCover, GeomSanFranciscoPoly) {
+    auto sanFranciscoGeom = Polygon<double>{
+        {
+            {-122.5143814086914,37.779127216982424},
+            {-122.50811576843262,37.72721239056709},
+            {-122.50313758850099,37.70820178063929},
+            {-122.3938751220703,37.707454835665274},
+            {-122.37567901611328,37.70663997801684},
+            {-122.36297607421874,37.71343018466285},
+            {-122.354736328125,37.727280276860036},
+            {-122.36469268798828,37.73868429065797},
+            {-122.38014221191408,37.75442980295571},
+            {-122.38391876220702,37.78753873820529},
+            {-122.35919952392578,37.8065289741725},
+            {-122.35679626464844,37.820632846207864},
+            {-122.3712158203125,37.835276322922695},
+            {-122.3818588256836,37.82958198283902},
+            {-122.37190246582031,37.80788523279169},
+            {-122.38735198974608,37.791337175930686},
+            {-122.40966796874999,37.812767557570204},
+            {-122.46425628662108,37.807071480609274},
+            {-122.46803283691405,37.810326435534755},
+            {-122.47901916503906,37.81168262440736},
+            {-122.48966217041016,37.78916666399649},
+            {-122.50579833984375,37.78781006166096},
+            {-122.5143814086914,37.779127216982424}
+        }
+    };
+
+    EXPECT_EQ((std::vector<UnwrappedTileID>{ { 10, 163, 395 }, { 10, 163, 396 } }),
+        util::tileCover(sanFranciscoGeom, 10));
+
+    EXPECT_EQ((std::vector<UnwrappedTileID>{
+            { 12, 654, 1582 }, { 12, 655, 1582 },
+            { 12, 654, 1583 },  { 12, 655, 1583 },
+            { 12, 654, 1584 }, { 12, 655, 1584 }
+        }),
+        util::tileCover(sanFranciscoGeom, 12));
+}
+
+TEST(TileCover, GeomSpiky) {
+    auto spikyGeom = Polygon<double> {
+        {
+            {16.611328125,8.667918002363134},
+            {13.447265624999998,3.381823735328289},
+            {15.3369140625,-6.0968598188879355},
+            {16.7431640625,1.0546279422758869},
+            {18.193359375,-10.314919285813147},
+            {19.248046875,-1.4061088354351468},
+            {20.698242187499996,-4.565473550710278},
+            {22.587890625,0.3515602939922709},
+            {24.2138671875,-11.73830237143684},
+            {29.091796875,5.003394345022162},
+            {26.4990234375,9.752370139173285},
+            {26.0595703125,7.623886853120036},
+            {24.9169921875,9.44906182688142},
+            {22.587890625,6.751896464843375},
+            {21.665039062499996,12.597454504832017},
+            {20.9619140625,8.189742344383703},
+            {18.193359375,14.3069694978258},
+            {16.611328125,8.667918002363134}
+        }
+    };
+
+    //These are all different from the js tile-cover lib
+    EXPECT_EQ(1742u, util::tileCover(spikyGeom, 10).size());
```

Comment:
I suspect that this test has different results based on floating point precision difference between the JS and CPP implementations. It's hard to reproduce with a small test case.